contributing/development/core_and_modules/core_types.rst

```diff
@@ -103,7 +103,18 @@ which are equivalent to new, delete, new[] and delete[].
 memnew/memdelete also use a little C++ magic and notify Objects right
 after they are created, and right before they are deleted.
 
-For dynamic memory, use Vector<>.
+For dynamic memory, use Godot's ``Vector<>`` or one of its variations.
+Godot's ``Vector<>`` behaves much like an STL ``std::vector<>``, but is simpler,
+thread safe, and uses Copy-On-Write semantics.
+It can be safely passed via public API.
+
+The ``Packed*Array`` :ref:`types <doc_gdscript_packed_arrays>` are aliases for
+specific ``Vector<*>`` types (e.g., ``PackedByteArray``, ``PackedInt32Array``)
+that are accessible via GDScript. Prefer using the ``Packed*Array`` aliases
```

Comment:
I haven't seen this be done very much, is using `Packed` aliases actually recommended?

Reply:
`core` might be the exception, as the aliases are defined in `Variant`. Virtually everything outside of `core` will include `Variant` by default, so in those instances I've generally seen the aliases used more.

Reply:
My personal interpretation is to use `Packed*Array` for functions exposed to scripts, and `Vector<>` for other occasions.